unit-tests-java8/src/test/java/org/eclipse/collections/impl/list/mutable/RandomAccessListAdapterTest.java

```diff
@@ -27,4 +33,30 @@ public final <T> MutableList<T> newWith(T... elements)
     {
         return new RandomAccessListAdapter<>(new ArrayList<>(Arrays.asList(elements)));
     }
+
+    @Override
+    public void List_subList_subList_clear()
+    {
+        List<String> list = this.newWith("A", "B", "C", "D", "E", "F");
+        List<String> sublist = list.subList(3, 6);
+        List<String> sublist2 = sublist.subList(0, 2);
+        assertEquals(Lists.immutable.with("D", "E", "F"), sublist);
+        assertEquals(Lists.immutable.with("D", "E"), sublist2);
+
+        sublist2.clear();
+        assertEquals(Lists.immutable.with("A", "B", "C", "F"), list);
+        assertEquals(Lists.immutable.with("F"), sublist);
+        assertEquals(Lists.immutable.with(), sublist2);
+
+        sublist2.add("J");
+        assertEquals(Lists.immutable.with("A", "B", "C", "J", "F"), list);
+        assertEquals(Lists.immutable.with("J", "F"), sublist);
+        assertEquals(Lists.immutable.with("J"), sublist2);
+
+        sublist.add("H");
+        assertEquals(Lists.immutable.with("A", "B", "C", "J", "F", "H"), list);
+        assertEquals(Lists.immutable.with("J", "F", "H"), sublist);
+
+        assertThrows(ConcurrentModificationException.class, sublist2::size);
```

Comment:
Our `mutableList` don't have `modCount` field that is used to provides fail-fast behavior defined in `java.util.AbstractList`.  

Back to my test code, `subList2` is a subList of `sublist`, and `sublist` is a subList of `list`.  

when I call `sublist.add("H");` , it can update the `modCount` of `sublist` and `list`, but can't update the `modCount` of `sublist`. Because sublist2 is a instance of `java.util.ArrayList.SubList`, it can trigger ` checkForComodification()` when we call every method it has. The `modCount` of sublist2 isn't equals to The `modCount` of it's root. As a result , `ConcurrentModificationException` is thrown.

Reply:
I'm wondering why you included this in the test. I don't think it's a new behavior change, just missing test coverage. Is that right?